### PR TITLE
SONARXML-52 upgrade to SonarQube API 6.2

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java [hunk 1]
```diff
@@ -99,62 +84,55 @@ private void runChecks(XmlFile xmlFile) {
...
 
-  private void saveSyntaxHighlighting(List<HighlightingData> highlightingDataList, InputFile inputFile) {
-    Highlightable highlightable = perspective(Highlightable.class, inputFile);
-    if (highlightable != null) {
-      HighlightingBuilder highlightingBuilder = highlightable.newHighlighting();
-
-      for (HighlightingData highlightingData : highlightingDataList) {
-        highlightingBuilder.highlight(highlightingData.startOffset(), highlightingData.endOffset(), highlightingData.highlightCode());
-      }
+  private void saveSyntaxHighlighting(SensorContext context, List<HighlightingData> highlightingDataList, InputFile inputFile) {
```
Comment (SonarTech) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java]: MAJOR Make "saveSyntaxHighlighting" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java [hunk 2]
```diff
@@ -99,62 +84,55 @@ private void runChecks(XmlFile xmlFile) {
...
+    NewHighlighting highlighting = context.newHighlighting().onFile(inputFile);
 
-      highlightingBuilder.done();
+    for (HighlightingData highlightingData : highlightingDataList) {
+      highlighting.highlight(highlightingData.startOffset(), highlightingData.endOffset(), highlightingData.highlightCode());
```
Comment (SonarTech) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java]: MINOR Remove this use of "highlight"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java [hunk 3]
```diff
@@ -20,22 +20,22 @@
...
 import com.google.common.annotations.VisibleForTesting;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.sonar.api.batch.Sensor;
-import org.sonar.api.batch.SensorContext;
+import java.util.List;
 import org.sonar.api.batch.fs.FilePredicate;
 import org.sonar.api.batch.fs.FileSystem;
 import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.TextRange;
```
Comment (SonarTech) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java]: MINOR Remove this unused import 'org.sonar.api.batch.fs.TextRange'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java [hunk 4]
```diff
@@ -20,22 +20,22 @@
...
+import org.sonar.api.batch.fs.internal.DefaultTextPointer;
```
Comment (SonarTech) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java]: MINOR Remove this unused import 'org.sonar.api.batch.fs.internal.DefaultTextPointer'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java [hunk 5]
```diff
@@ -20,22 +20,22 @@
...
+import org.sonar.api.batch.fs.internal.DefaultTextRange;
```
Comment (SonarTech) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java]: MINOR Remove this unused import 'org.sonar.api.batch.fs.internal.DefaultTextRange'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 8 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [XmlPlugin.java#L37](https://github.com/SonarSource/sonar-xml/blob/0af48157c296c41a8ca78f2791f559d3879d885f/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlPlugin.java#L37): Remove this use of "SonarPlugin"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [IllegalTabCheck.java#L43](https://github.com/SonarSource/sonar-xml/blob/0af48157c296c41a8ca78f2791f559d3879d885f/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/IllegalTabCheck.java#L43): Remove this use of "SubCharacteristics"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [IndentCheck.java#L44](https://github.com/SonarSource/sonar-xml/blob/0af48157c296c41a8ca78f2791f559d3879d885f/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/IndentCheck.java#L44): Remove this use of "SubCharacteristics"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [NewlineCheck.java#L42](https://github.com/SonarSource/sonar-xml/blob/0af48157c296c41a8ca78f2791f559d3879d885f/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/NewlineCheck.java#L42): Remove this use of "SubCharacteristics"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [XmlPlugin.java#L37](https://github.com/SonarSource/sonar-xml/blob/c9604df5d469ae95c6564761fe7d6c49d446c2c7/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlPlugin.java#L37): Remove this use of "SonarPlugin"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [IllegalTabCheck.java#L43](https://github.com/SonarSource/sonar-xml/blob/c9604df5d469ae95c6564761fe7d6c49d446c2c7/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/IllegalTabCheck.java#L43): Remove this use of "SubCharacteristics"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [IndentCheck.java#L44](https://github.com/SonarSource/sonar-xml/blob/c9604df5d469ae95c6564761fe7d6c49d446c2c7/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/IndentCheck.java#L44): Remove this use of "SubCharacteristics"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [NewlineCheck.java#L42](https://github.com/SonarSource/sonar-xml/blob/c9604df5d469ae95c6564761fe7d6c49d446c2c7/sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/NewlineCheck.java#L42): Remove this use of "SubCharacteristics"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by janos-ss** [comment]
After merging this, it could be a good idea to merge Tom's PR too about the licence and copyright files. https://github.com/SonarSource/sonar-xml/pull/37

**Pull request merged**